plugin.program.autowidget/resources/lib/refresh.py

```diff
@@ -269,54 +316,49 @@ def refresh_paths(notify=False, force=False):
     return True, "AutoWidget"
 
 
-def get_files_list(path, widget_id=None):
-    hash = utils.path2hash(path)
-    _, files, _ = utils.cache_expiry(hash, widget_id)
+def get_files_list(path, label=None, widget_id=None, background=True):
+    hash = cache.path2hash(path)
+    _, files, _ = cache.cache_expiry(path, widget_id, background=background)
     if files is None:
-        # We had no old content so have to block and get it now
+        # Should only happen now when background is False
         utils.log("Blocking cache path read: {}".format(hash[:5]), "info")
-        files, changed = utils.cache_files(path, widget_id)
+        files, changed = cache.cache_files(path, widget_id)
 
     new_files = []
     if "result" in files:
         files = files.get("result", {}).get("files", [])
-        if not files:
-            utils.log("No items found for {}".format(path))
-            return [], hash
-
-        for file in files:
-            new_file = {k: v for k, v in file.items() if v is not None}
-
-            if "art" in new_file:
-                for art in new_file["art"]:
-                    new_file["art"][art] = utils.clean_artwork_url(file["art"][art])
-            if "cast" in new_file:
-                for idx, cast in enumerate(new_file["cast"]):
-                    new_file["cast"][idx]["thumbnail"] = utils.clean_artwork_url(
-                        cast.get("thumbnail", "")
-                    )
-            new_files.append(new_file)
-
-        return new_files, hash
     elif "error" in files:
-        os.remove(os.path.join(_addon_data, "{}.cache".format(hash)))
-        utils.log("Invalid cache file removed for {}".format(hash))
-        return None, hash
-    else:
         utils.log("Error processing {}".format(hash), "error")
-        return None, hash
+        error_tile = utils.make_holding_path(
+            utils.get_string(30139).format(label), "alert", hash=hash
+        )
+        files = error_tile.get("result", {}).get("files", [])
+        cache_path = os.path.join(_addon_data, "{}.cache".format(hash))
+        if os.path.exists(cache_path):
+            os.remove(cache_path)
+        utils.log("Invalid cache file removed for {}".format(hash))
+
+    if not files:
+        utils.log("No items found for {}".format(hash))
+        empty_tile = utils.make_holding_path(

```

Comment:
If the cache is valid, but there are no files to return, then we make an "empty path" tile instead.